src/c_promenne.md

```diff
@@ -1,53 +1,119 @@
+semestru se s nimi budeme setkávat neustále. Není tak náhodou, že jedním z nejzákladnějších příkazů
+v C je právě vytvoření proměnné. Tím řekneme počítači, aby vyčlenil (tzv. **naalokoval**) místo v paměti,
+které si v programu nějak pojmenujeme a dále se na něho pomocí jeho jména můžeme odkazovat.[^1]
 
-Proměnnou vytvoříme jednoduše tak, že nadefinujeme její typ a pak její
-jméno, případně můžeme na jednom řádku definovat více proměnných jednoho
-typu různých jmen oddělených čárkou.
+[^1]: O tom, jak přesně tato alokace paměti probíhá, se dozvíte později v sekci o
+[ukazatelích](c_ukazatele.md).
 
+### Deklarace a platnost
+Takto vypadá **deklarace** (vytvoření) jednoduché proměnné s názvem `age`:
 ```c
-int counter;
-int sum, no_elements = 0;
+int age;
 ```
+Jakmile proměnnou nadeklarujeme, tak z ní můžeme buď číst anebo zapisovat paměť, kterou tato proměnná
+reprezentuje, pomocí jejího názvu (zde `age`).
 
-Na výše uvedeném výpisu je použit výsek kódu, který deklaruje na prvním
-řádku celočíselnou proměnnou typu `int`, která reprezentuje integer, což je
-datový typ reprezentující celá čísla.
-Druhý řádek demonstruje, že je možno
-deklarovat více proměnných jednoho datového typu na jednom řádku. Je
-deklarována proměnná `sum` a `no_elements`, do které je přiřazena
-počáteční hodnota `0` (pokud do proměnné rovnou přiřadíme hodnotu, je to definice proměnné).
-Pokud hodnotu do proměnné takto nepřiřadíme na začatku při její deklaraci,
-musíme takovou proměnnou před prvním použitím nejříve nastavit na
-požadovanou hodnotu. Typicky se toto děje, pokud používáme proměnnou
-jako nějaké počítadlo.
-
+Proměnná je platná (lze ji používat) vždy od místa
+deklarace do konce **bloku**, ve kterém byla nadeklarována. Bloky jsou kusy kódu ohraničené složenými
+závorkami (`{` a `}`):
 ```c
-int counter;
+int main() {
+    int a;
+
+    {
+        // zde je platné pouze `a`
+        int b;
+        // zde je platné `a` i `b`
+    } // zde končí platnost proměnné `b`
 
-int sum, no_elements = 0;
+    // zde je platné pouze `a`
 
-printf( "sum: %d\n", sum );                  // muze byt cokoli: sum: 432749
-printf( "no_elements: %d\n", no_elements );  // no_elements: 0
+    return 0;
+} // zde končí platnost proměnné `a`
 ```
 
-Na řádku 5 a 6 vidíme použití knihovní funkce `printf`, která nám bude sloužit pro
-tisk obsahu proměnných na konzoli. Můžete vidět, že funkce `printf` používá
-formátovací řetězec, kde celá čísla (`int`eger) jsou zastoupena znakem `%d`.
-Obecně je formátovácí řetězec uvozen znakem `%` následovaný formátovacím
-znakem. Více o formátování výstupu se můžete dozvědet např. zde:
-[^1],[^2].
+### Datový typ
+`int` před názvem proměnné udává její datový typ, o kterém pojednává [následující sekce](c_datove_typy.md).
+Prozatím si řekněme, že `int` je zkratka pro `integer`, tedy celé číslo. Tím říkáme programu, že má
+tuto proměnnou (resp. paměť, kterou proměnná reprezentuje) interpretovat jako číslo.
```

Comment:
mozna zminit ze to cislo neni nekonecne velke?

Reply:
To bude asi stačit v té sekci datové typy.